Src/ElasticScale.Client/ShardManagement/SqlStore/SqlShardMapManagerCredentials.cs

```diff
@@ -121,8 +122,10 @@ public string ShardMapManagerLocation
         /// <param name="parameterName">Parameter name of the connection string object.</param>
         internal static void EnsureCredentials(SqlConnectionStringBuilder connectionString, string parameterName)
         {
-            // Check for integrated authentication.
-            if (!connectionString.IntegratedSecurity)
+            // Check for integrated authentication or active directory integrated authentication (if supported)
+            if (!(connectionString.IntegratedSecurity || 
+                 (ShardMapUtils.IsActiveDirectoryAuthenticationSupported && 
+                  connectionString[ShardMapUtils.Authentication].ToString().Equals(ShardMapUtils.ActiveDirectoryIntegratedStr))))
```

Comment:
Yes, all callers first check ShardMapUtils.ValidateAuthenticationInConnectionString() before calling this function, so I can remove ShardMapUtils.IsActiveDirectoryAuthenticationSupported check.